monoio-http/src/common/macros.rs

```diff
@@ -0,0 +1,107 @@
+#[macro_export]
+macro_rules! impl_cookie_extractor {
+    ($struct:ident, $cookie_str:literal) => {
+        impl<P> $struct<P> {
+            /// Builds a cookie jar from the $struct's headers.
+            pub fn build_cookie_jar(&mut self) -> Result<CookieJar, ExtractError> {
+                let headers = self.inner.headers();
+                let mut jar = CookieJar::new();
+
+                if let Some(cookie_header) = headers.get($cookie_str) {
+                    let cookie_str = cookie_header
+                        .to_str()
+                        .map_err(|_| ExtractError::InvalidHeaderValue)?;
+                    let cookie_owned_str = cookie_str.to_string();
+                    println!("{}", cookie_owned_str);
+                    for cookie in Cookie::split_parse(cookie_owned_str) {
```

Comment:
Maybe add a wrapper to cache `Cookie::split_parse` result is enough(or construct a hashmap).